source/common/http/conn_manager_impl.cc

```diff
@@ -901,8 +901,11 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
       request_headers_->Expect()->value() == Headers::get().ExpectValues._100Continue.c_str()) {
     // Note in the case Envoy is handling 100-Continue complexity, it skips the filter chain
     // and sends the 100-Continue directly to the encoder.
-    chargeStats(continueHeader());
-    response_encoder_->encode100ContinueHeaders(continueHeader());
+    const auto& continue_headers = continueHeader();
+    const auto status = response_encoder_->encode100ContinueHeaders(continue_headers);
+    // We must make sure static `continue_headers` is valid in its content and encoding not fail.
+    RELEASE_ASSERT(status.ok(), "failed to encode 100-continue headers");
```

Comment:
fixed in https://github.com/envoyproxy/envoy/pull/15658/commits/e8fd6d9d1c130ceee9a0711491cb42cef0e2e698  cc @asraa since you are in favor of RELEASE_ASSERT?